components/MapComponent.tsx

```diff
@@ -1,123 +1,146 @@
-import React, { useState } from "react";
-import { Button, StyleProp, View, ViewStyle } from "react-native";
-import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";
+import { Location } from "lib/location/Location"
+import React, { useState } from "react"
+import { StyleProp, View, ViewStyle } from "react-native"
+import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps"
+
+// Type for the map markers, in order to separate their logic out
+export interface mapMarker {
+  key: string
+  title: string
+  location: Location
+}
+
+interface Props<T> {
+  // Contain style specifically for style on container for map
+  // Map style specifically for style on map
+  containStyle?: StyleProp<ViewStyle>
+  mapStyle?: StyleProp<ViewStyle>
 
-interface Props {
-  containStyle?: StyleProp<ViewStyle>;
-  mapStyle?: StyleProp<ViewStyle>;
   initialRegion: {
-    latitude: number,
-    longitude: number,
-    latitudeDelta: number,
-    longitudeDelta: number,
+    latitude: number
+    longitude: number
+    latitudeDelta: number
+    longitudeDelta: number
   }
-  markers: {
-  key: number,
-  place: string,
-  lat: number,
-  lng: number,
-  pinColor: string
-  }[];
+
+  markers: mapMarker[]
+
   movementSettings: {
-  canScroll: boolean,
-  canZoom: boolean,
-  canRotate: boolean,
-  };
+    canScroll: boolean
+    canZoom: boolean
+    canRotate: boolean
+  }
+
+  extractKey: (event: any) => string
```

Comment:
This shouldn't be necessary if you already require a key in the `mapMarker` type.